util/lua/lua_test.go

```diff
@@ -954,3 +954,12 @@ func createMockResource(kind string, name string, replicas int) *unstructured.Un
             image: nginx
     `, kind, name, replicas))
 }
+
+func Test_getHealthScriptPaths(t *testing.T) {
```

Comment:
Multiple new codepath are not covered by tests, such as `getWildcardBuiltInHealthOverrideLua` and the changes in `GetHealthScript`

Reply:
The new functions and the changes in GetHealthScript are covered by existing tests: https://app.codecov.io/gh/argoproj/argo-cd/pull/22919?dropdown=coverage&src=pr&el=continue&utm_medium=referral&utm_source=github&utm_content=comment&utm_campaign=pr+comments&utm_term=argoproj

The only lines that aren't covered are error handling. I'll see if I can get some more coverage on those.

Reply:
I went through each of the error cases, and none of them should be possible unless we add an intentionally messed-up directory to resource_customizations. And if that happens, the problem should be caught while trying to implement the new test anyway. I don't think there's any point synthesizing that scenario.